File: Behavioral/Lab14/C#/Builders/RoomBuilder.cs
using System;
using System.Collections.Generic;
using System.Text;
using C_.Entities;

namespace C_.Builders
{
    public class RoomBuilder : EstateBuilder
    {
        public override void Reset(){
            _estate = new Room();
        }
        public RoomBuilder SetNeighbors(bool goodOrBad = false){
            var h = (Room)_estate;
            h.IsGoodNeighbors = goodOrBad;
            return this;
        }
        public RoomBuilder SetHeatingType(string heating){
            var h = (Room)_estate;
            h.HeatingType = heating;
            return this;
        }
    }
}

File: Structural/Lab10/Typescript/src/Builders/room_builder.ts
import { EstateBuilder } from './estate_builder';
import { Room } from '../Entities/room';

export class RoomBuilder extends EstateBuilder {
    public reset(): void {
        this._estate = new Room();
    }

    public setNeighbors(goodOrBad: boolean = false): RoomBuilder {
        const a = this._estate as Room;
        a.isGoodNeighbors = goodOrBad;
        return this;
    }

    public setHeatingType(type: string): RoomBuilder {
        const a = this._estate as Room;
        a.heatingType = type;
        return this;
    }
}
